tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -265,7 +270,14 @@ def _fit(self,
                 with self.model.no_logging():
                     original_path = self.model.path
                     self.model.path = Path(temp_dir)
-                    self.model.fit_one_cycle(params['epochs'], params['lr'], cbs=callbacks)
+
+                    len_val = len(X_val) if X_val is not None else 0
+                    epochs = self._get_epochs_number(samples_num=len(X) + len_val, epochs=params['epochs'], batch_size=batch_size, time_left=time_left)
+                    if epochs == 0:
+                        # Stop early if there is not enough time to train a full epoch
+                        raise TimeLimitExceeded
+
+                    self.model.fit_one_cycle(epochs, params['lr'], cbs=callbacks)
```

Comment:
On line 293, add a new line:

```
self.params_trained['epochs'] = epochs
```

This will ensure refit_full works properly.

Reply:
done
```
0               {'layers': None, 'emb_drop': 0.1, 'ps': 0.1, 'bs': 'auto', 'lr': 0.01, 'epochs': 'auto', 'early.stopping.min_delta': 0.0001, 'early.stopping.patience': 20, 'smoothing': 0.0}   {'epochs': 8, 'best_epoch': 4}                  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': ['object'], 'get_features_kwargs': None, 'get_features_kwargs_extra': None}                                    [NeuralNetFastAI_BAG_L1]                                                              [WeightedEnsemble_L3]  
1                                                                                                                                                                      {'ensemble_size': 100}             {'ensemble_size': 1}  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': None, 'get_features_kwargs': None, 'get_features_kwargs_extra': None, 'drop_unique': False}            [NeuralNetFastAI_BAG_L1, NeuralNetFastAI_BAG_L2]                                                                                 []  
2               {'layers': None, 'emb_drop': 0.1, 'ps': 0.1, 'bs': 'auto', 'lr': 0.01, 'epochs': 'auto', 'early.stopping.min_delta': 0.0001, 'early.stopping.patience': 20, 'smoothing': 0.0}  {'epochs': 13, 'best_epoch': 9}                  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': ['object'], 'get_features_kwargs': None, 'get_features_kwargs_extra': None}                                                          []                 [NeuralNetFastAI_BAG_L2, WeightedEnsemble_L3, WeightedEnsemble_L2]  
3                                                                                                                                                                      {'ensemble_size': 100}             {'ensemble_size': 1}  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': None, 'get_features_kwargs': None, 'get_features_kwargs_extra': None, 'drop_unique': False}                                    [NeuralNetFastAI_BAG_L1]                                                                                 []  
4                                                                                                                                                    {'ensemble_size': 100, 'weights': [1.0]}                               {}  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': None, 'get_features_kwargs': None, 'get_features_kwargs_extra': None, 'drop_unique': False}  [NeuralNetFastAI_BAG_L2_FULL, NeuralNetFastAI_BAG_L1_FULL]                                                                                 []  
5                                                                                                                                                    {'ensemble_size': 100, 'weights': [1.0]}                               {}  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': None, 'get_features_kwargs': None, 'get_features_kwargs_extra': None, 'drop_unique': False}                               [NeuralNetFastAI_BAG_L1_FULL]                                                                                 []  
6   {'layers': None, 'emb_drop': 0.1, 'ps': 0.1, 'bs': 'auto', 'lr': 0.01, 'epochs': 8, 'early.stopping.min_delta': 0.0001, 'early.stopping.patience': 20, 'smoothing': 0.0, 'best_epoch': 4}   {'epochs': 8, 'best_epoch': 4}                  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': ['object'], 'get_features_kwargs': None, 'get_features_kwargs_extra': None}                               [NeuralNetFastAI_BAG_L1_FULL]                                                         [WeightedEnsemble_L3_FULL]  
7  {'layers': None, 'emb_drop': 0.1, 'ps': 0.1, 'bs': 'auto', 'lr': 0.01, 'epochs': 13, 'early.stopping.min_delta': 0.0001, 'early.stopping.patience': 20, 'smoothing': 0.0, 'best_epoch': 9}  {'epochs': 13, 'best_epoch': 9}                  {'max_memory_usage_ratio': 1.0, 'max_time_limit_ratio': 1.0, 'max_time_limit': None, 'min_time_limit': 0, 'ignored_type_group_special': None, 'ignored_type_group_raw': ['object'], 'get_features_kwargs': None, 'get_features_kwargs_extra': None}                                                          []  [NeuralNetFastAI_BAG_L2_FULL, WeightedEnsemble_L3_FULL, WeightedEnsemble_L2_FULL]  
```